app/src/lib/git/merge.ts

```diff
@@ -9,15 +13,73 @@ export async function merge(
   await git(['merge', branch], repository.path, 'merge')
 }
 
+/**
+ * Find the base commit between two refs
+ *
+ * @returns the commit id of the merge base, or null if the two refs do not
+ *          have a common base
+ */
 export async function getMergeBase(
   repository: Repository,
   firstRef: string,
   secondRef: string
-): Promise<string> {
+): Promise<string | null> {
   const process = await git(
     ['merge-base', firstRef, secondRef],
     repository.path,
-    'merge-base'
+    'merge-base',
+    {
+      successExitCodes: new Set([0, 1]),
```

Comment:
What do these exit codes mean? Is it possible to pull these codes into constant so we don't have to throw around magic numbers?

Reply:
The docs aren't really clear on what exit codes are expected here but from my testing showed `0` means a common ancestor was found. `1` means one could not be found (no `stdout` or `stderr` occurred).

Given the rest of the function is rather boring does introducing constants here help readability, or is there something else we can do here?

Reply:
I think it's a readability improvement. If we can pass the exit codes we care about to the `git` function, I think we should always use a name instead of numbers. I've seen this in a few places now, so I figured passing these codes will become more common in the future.

Reply:
>  I've seen this in a few places now, so I figured passing these codes will become more common in the future.

Are we talking about `expectedErrors` (where an error message matches a known regex, and we map this to an enum in code) rather than `successExitCodes` (the raw numbers)?

I'm definitely in favour of preferring `expectedErrors` where it makes sense, but we don't get any `stdout` or `stderr` in this case to indicate anything went wrong - just the `1` exit code. [Here's the relevant Git source code, for reference](https://github.com/git/git/blob/2f743933341f276111103550fbf383a34dfcfd38/builtin/merge-base.c#L11-L28).